fvtest/compilertriltest/ShiftAndRotateTest.cpp

```diff
@@ -1188,7 +1215,9 @@ TEST_P(UInt64MaskThenShift, UsingLoadParam) {
 
 INSTANTIATE_TEST_CASE_P(ShiftAndRotateTest, UInt64MaskThenShift, ::testing::Combine(
     ::testing::ValuesIn(static_cast< std::vector<std::tuple<std::tuple<uint64_t, uint64_t>, int32_t>> (*) (void)>(test_mask_then_shift_values)()),
-    ::testing::Values(std::make_tuple<const char*, uint64_t (*) (uint64_t, uint64_t, int32_t)>("lushr", static_cast<uint64_t (*) (uint64_t, uint64_t, int32_t)>(mask_then_shift_right))
+    ::testing::Values(
+        std::make_tuple<const char*, uint64_t (*) (uint64_t, uint64_t, int32_t)>("lushr", static_cast<uint64_t (*) (uint64_t, uint64_t, int32_t)>(mask_then_shift_right)),
+        std::make_tuple<const char*, uint64_t (*) (uint64_t, uint64_t, int32_t)>("lushl", static_cast<uint64_t (*) (uint64_t, uint64_t, int32_t)>(mask_then_shift_left))
```

Comment:
It seems the tests here are now a superset of the tests in LongAndAsRotateTest. Can we now deprecate that test bucket so we don't have duplication?

[1] https://github.com/eclipse/omr/blob/master/fvtest/compilertriltest/LongAndAsRotateTest.cpp

Reply:
It seems as if all of the tests in `LongAndAsRotateTest` are a subset of the tests added here. Should I just be removing the file entirely?

Reply:
If `LongAndAsRotateTest` is a subset of `ShiftAndRotateTest` all that needs to be done is to remove the file.

Reply:
I'm not sure I agree that `LongAndAsRotateTest` is now a subset of `ShiftAndRotateTest`... It looks to me like `LongAndAsRotateTest` contains test cases that test mixes of conversions and masks, while the new test cases added in `ShiftAndRotateTest` are testing mixes of masks and shifts. While these do trigger similar optimizations, I don't really see any duplication between the two...

Reply:
@AidanHa can you comment on the above?

Reply:
I think the new tests are testing the `LongAndAsRotate` tests with a shift of `zero`. However, it does not test for conversions, so maybe keeping the `LongAndAsRotate` tests for test for conversions might be the better choice.

Reply:
Opened up #4473 to track this.